sqlmesh/core/engine_adapter/redshift.py

```diff
@@ -128,7 +128,16 @@ def _fetch_native_df(
     ) -> pd.DataFrame:
         """Fetches a Pandas DataFrame from the cursor"""
         self.execute(query, quote_identifiers=quote_identifiers)
-        return self.cursor.fetch_dataframe()
+
+        # We manually build the `DataFrame` here because the driver's `fetch_dataframe`
+        # method does not respect the active case-sensitivity configuration.
+        #
+        # Context: https://github.com/aws/amazon-redshift-python-driver/issues/238
+        fetcheddata = self.cursor.fetchall()
+        columns = [column[0] for column in self.cursor._getDescription() or []]
+
+        result = [tuple(column for column in rows) for rows in fetcheddata]
```

Comment:
is rows not already a tuple?

could you call tuple(rows) ?

Reply:
I'll check, I did copy the driver's [implementation](https://github.com/aws/amazon-redshift-python-driver/blob/64cbd54ef6a5e71d98ce193630d09267cc154379/redshift_connector/cursor.py#L535) here, so I didn't double-check if it can be further simplified.

Reply:
You're right, good catch. Put a breakpoint to verify:

```
>>> fetcheddata
([(1, 'foo', '{"bar": "bla", "boBa":"val", "someId":"someVal", "foo": "2024-10-14T10:00"}', '2024-10-14 10:00', 'someVal', 'val', datetime.datetime(2024, 10, 14, 10, 0))],)
>>> [tuple(column for column in rows) for rows in fetcheddata]
[(1, 'foo', '{"bar": "bla", "boBa":"val", "someId":"someVal", "foo": "2024-10-14T10:00"}', '2024-10-14 10:00', 'someVal', 'val', datetime.datetime(2024, 10, 14, 10, 0))]
>>> [tuple(rows) for rows in fetcheddata]
[(1, 'foo', '{"bar": "bla", "boBa":"val", "someId":"someVal", "foo": "2024-10-14T10:00"}', '2024-10-14 10:00', 'someVal', 'val', datetime.datetime(2024, 10, 14, 10, 0))]
```